src/main/java/org/occiware/tinom/aggregators/OrderedWeightedAveraging.java

```diff
@@ -0,0 +1,177 @@
+        double t = Math.max(t_m, t_p);
+
+        // System.out.println("t = " + t);
+
+        double[] w = new double[n];
+        double s = 0.0;
+        for (int i = 0; i < n; i++) {
+            w[i] = Math.pow(t, i);
+            s += w[i];
+        }
+        for (int i = 0; i < n; i++) {
+            w[i] /= s;
+        }
+
+        return new OrderedWeightedAveraging(w);
+    }
+
+
+    public final double[] weights;
+
+    public OrderedWeightedAveraging(double... weights) {
+        this.weights = weights;
+        double sum = 0.0;
+        for (double w : weights)
+            sum += w;
+        double diff = Math.abs(1.0 - sum);
+        if (diff > 0.0001)
+            throw new IllegalArgumentException(String.format("sum(weights) must be 1.0 (was: %f, %.5f)", sum, diff));
+    }
+
+    @Override
+    public double apply(double... values) {
+        // we need same lenghts
+        int min = Math.min(values.length, weights.length);
+        // sort
+        Arrays.sort(values);
+        OrderedWeightedAveraging.reverse(values);
+        //
+        double s = 0.0;
+        for (int i = 0; i < min; i++)
+            s += values[i] * weights[i];
+
+        return s;
+    }
+
+    /**
+     * Calculate the orness of this OWA operator.
+     *
+     * To get the andness use it's definition: <code>double andness = 1.0 - orness();</code>
+     *
+     * @return the orness [0, 1]
+     */
+    public double orness() {
+        double n = weights.length;
+        double s = 0.0;
+        for (int i = 0; i < weights.length; i++)
+            s += (n - (i+1)) * weights[i];
+        return s / (n - 1.0);
+    }
+
+    /**
+     * The dual of orness. See {@link #orness()}
+     * @return
+     */
+    public double andness() {
+        return 1.0 - orness();
+    }
+
+    /**
+     * Calculate the relative dispersion (entropy) of the OWA operator.
+     *
+     * @return the dispersion [0, ln n]
+     */
+    public double dispersion() {
+        double n = weights.length;
+        double s = 0.0;
+        for (double w : weights)
+            if (w > 0.0)
+                s += w * Math.log(1.0 / w);
+        return s / Math.log(n);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < weights.length; i++)
+            sb.append(" " + weights[i]);
+        return "OWA{" +
+                "weights=" + sb.substring(1).toString() +
+                '}';
+    }
+
+    /**
+     * Reverse an array.
+     *
+     * @param array the array
+     */
+    public static void reverse(double[] array) {
+        if (array == null) {
+            return;
+        }
+        int i = 0;
+        int j = array.length - 1;
+        double tmp;
+        while (j > i) {
+            tmp = array[j];
+            array[j] = array[i];
+            array[i] = tmp;
+            j--;
+            i++;
+        }
+    }
+
```

Comment:
Why all these line breaks at the end?
This is not a major issue though, it's just ugly.